app/plugin_api.go

```diff
@@ -57,6 +63,39 @@ func (api *PluginAPI) LoadPluginConfiguration(dest interface{}) error {
 	}
 }
 
+func (api *PluginAPI) ValidateConfig(config map[string]interface{}) bool {
```

Comment:
3/5 should return an `error` not `bool`

Reply:
The issue I have with this, what if the plugin is missing many required configuration keys. would you consider a multi error setup such as this one
```go
	type errorGroup interface {
		// Returns a slice containing the underlying list of errors.
		//
		// This slice MUST NOT be modified by the caller.
		Errors() []error
	}
```
I found this here within [uber-go/multierr](https://github.com/uber-go/multierr/blob/master/error.go)

Reply:
Actually fixed the problem of returning many missing required configuration keys by creating a custom error type
``` go
type ConfigError struct {
	PluginId    string
	MissingKeys []string
}
```